pages/index.tsx

```diff
@@ -238,7 +218,7 @@ export default function Home() {
                     </div>
                 </div>
                 <div className="py-3 py-md-5">
-                    {currentAndUpcomingHackathons ? (
+                    {currentAndUpcomingHackathons !== [] ? (
                         <>
                             {currentAndUpcomingHackathons.length ? (
```

Comment:
Empty array and `0` length array are the same. We need to check if we are loading the hackathons from the API here.

Reply:
what do you want me to do?

Reply:
```ts
var currentAndUpcomingHackathons = false
```

The false here was meaning loading, I guess. By this change, the spinner is never being rendered. I can think of two possible ways to fix this:
1. We can add a state hook for loading (true while Axios returns or initially and false otherwise) and replace this line with that hook.
2. Change type of this variable to `boolean or array` and if `typeof(variableName) === 'boolean'` render the spinner.

Any solution will work fine.

Reply:
But keeping it false or putting empty array, wont the spinner render in either case as long as the variable has an empty array i.e. no hackathon inside it.

Reply:
Yeah, my bad. Perhaps, a state variable to store this behavior will be better for rendering.